core/interfaces/src/main/java/info/nightscout/interfaces/ApsMode.kt

```diff
@@ -0,0 +1,13 @@
+package info.nightscout.interfaces
+
+enum class ApsMode() {
+    OPEN,
+    CLOSED,
+    LGS,
+    UNDEFINED;
+
+    companion object {
+
+        fun fromString(stringValue: String?) = values().firstOrNull { it.name == (stringValue?.uppercase() ?: UNDEFINED.name) } ?: UNDEFINED
+    }
```

Comment:
`(stringValue?.uppercase() ?: UNDEFINED.name)` is not necessary
simply `stringValue?.uppercase()`  is enough

Reply:
Ooops, that was redundant. Simplified in next commit.